dimod/generators/binpacking.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import numpy as np
+
+from dimod.binary_quadratic_model import BinaryQuadraticModel
+from dimod.constrained import ConstrainedQuadraticModel
+from typing import Tuple
+
+__all__ = ['bin_packing']
+
+
+def bin_packing(num_items: int,
+                seed: int = 32,
+                weight_range: Tuple[int, int] = (10, 30),
+                ) -> ConstrainedQuadraticModel:
+    """Returns a constrained quadratic model encoding a bin packing problem
+
+    Given the number of items, the code generates a random bin packing problem
+    formulated as a Constrained Quadratic model. The weights for each item are
+    integers uniformly drawn from in the weight_range. The bin capacity is set
```

Comment:
Nice to have this as either "from the range specified by ``weight_range`` " or "from the weight range"
Similarly below, nicer to format as :math:`num_items * mean(weights) / 5` or to use :code:`` for params (``num_items``) and then text, just looks clearer in RTD descriptions